usecases/scenario_testrun_usecase.go

```diff
@@ -0,0 +1,69 @@
+package usecases
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/pure_utils"
+	"github.com/checkmarble/marble-backend/repositories"
+	"github.com/checkmarble/marble-backend/usecases/executor_factory"
+	"github.com/checkmarble/marble-backend/usecases/security"
+	"github.com/pkg/errors"
+)
+
+type ScenarioTestRunUsecase struct {
+	transactionFactory             executor_factory.TransactionFactory
+	executorFactory                executor_factory.ExecutorFactory
+	scenarioPublicationsRepository repositories.ScenarioPublicationRepository
+	enforceSecurity                security.EnforceSecurityScenario
+	clientDbIndexEditor            clientDbIndexEditor
```

Comment:
we're not using this in the end :-) (you probably have a linter warning displayed)
we should check similarly to usecases/scenario_publications_usecase.go lines 89 to 100 if a "data preparation" (AKA index creation) is required.
Then, we have two options for the actual creation:
- either we trigger an index creation from the app, and only then allow to click on "start a test run" (that's what we currently have when we publish a scenario)
- or we go a step further and instrument this within the test run lifecycle, using a task queue for the async update.
We can have a chat about this in the afternoon if you'd like. If it makes things easier, it's ok for me to add this in a separate PR but it should be taken into account before the feature is released in prod.

Reply:
for the moment, I'd prefer to add this in another PR.. I will keep this in mind for the release of the feature

Reply:
yes, as discussed